### DISCUSSION TITLE
Stop cgal remeshing holes in boolean operations

### DISCUSSION THREAD

**Discussion by TheBEllis** [libigl/libigl#2114] [Q&A]
Hello libIGL team hope you are well! I have a bit of a problem with the mesh boolean operations I cannot figure out how to solve. I have the 2D surface geometry of a coil, that has two prongs. I then have the 2D surface geometry of a cube which (sort of) encloses the coil. 

One of the faces of the cube is co-planar with the boundary surface of the coil prongs. Hopefully the first picture I've attached sort of helps show this. I have been trying to use the boolean mesh operations to merge these two together nicely. The reason for the co-planar-ness is that this geometry will eventually be used in a finite element analysis and the boundary conditions of said problem require those parts be co-planar at those points.  

Unfortunately, the results I'm currently getting using a boolean resolve aren't looking so great as can be seen in the second picture. It looks like there is a layer of facets over the uppermost parts of the coil. Sort of like cgal has detected there might be a hole there and has tried to nicely, yet inconveniently, mesh it all up. Is there any way you know of the stop this behavior? I have tried with finer and courser meshes but mesh resolution doesn't appear to change anything. The third picture shows the top of the coil in the original geometry, so you can see what it should sort of look like.

You might be wondering why I haven't been using the triangle wrapper for this task, as I have a bunch of points (the top most faces of the coil prongs), and would like to generate a mesh around them. I have done this and have successfully gotten the face of the cube which is co-planar with the coil bits, with the correct cutouts meshed very nicely. Unfortunately I also need the other 5 faces and triangle only takes 2D coordinates as inputs from what I can tell, so if I could do it using cgal that would be nice:) Sorry for the long question, and if I haven't explained something well please let me know! 

![translucent](https://user-images.githubusercontent.com/54721825/199317037-94e7a427-6996-4a36-ab60-65b8dffcf7e0.png)
![image](https://user-images.githubusercontent.com/54721825/199316493-780ec7a0-e768-472c-a8d1-4976d993637c.png)
![image](https://user-images.githubusercontent.com/54721825/199316892-0dd432f8-d229-4bce-8f5a-ef0b3c5f9d1d.png)

**Comment by alecjacobson** [comment]
Could you please include the models to test this?

My guess is that these inputs are not exactly co-planar.
